utils/misc/dbutils.go

```diff
@@ -19,15 +20,20 @@ func GetConnectionString(c *config.Config, componentName string) string {
 	port := c.GetInt("DB.port", 5432)
 	password := c.GetString("DB.password", "ubuntu") // Reading secrets from
 	sslmode := c.GetString("DB.sslMode", "disable")
+	idleTxTimeout := c.GetDuration("DB.IdleTxTimeout", 5, time.Minute)
```

Comment:
Should we keep this timeout limited to 120 Secs as from the graph in the desc max txn time seems to be 60-65 secs only? Keeping high value of timeout may still result in similar issues we are trying to solve(even-though for a smaller time)

Reply:
This timeout affects connections waiting for a client query within an open transaction for longer than the specified amount of time.

So if the client opens a transaction, runs a query, keeps the transaction open, it does something with the results and that "something" takes longer than 120 seconds, then the connection is going to be terminated. This would mean that the client will quite likely have to restart from the beginning.

I know the max in the graph shows as 60-65 but there might be edge cases where it could go above 120 I guess. Hard to be 100% sure without checking all metrics of all time and even then, code changes so you might experience that behaviour anyway at some point. This is the reason why 5 minutes seems safer to me and still reasonable enough not to cause incidents due to connections left alive forever.